Pod/Classes/CDTQQueryValidator.m

```diff
@@ -112,6 +136,102 @@ + (NSArray *)addImplicitEq:(NSArray *)andClause
     return [NSArray arrayWithArray:accumulator];
 }
 
++ (NSArray *)compressMultipleNotOperators:(NSArray *)clause
+{
+    NSMutableArray *accumulator = [NSMutableArray array];
+    
+    for (NSDictionary *fieldClause in clause) {
+        NSObject *predicate = nil;
+        NSString *fieldName = nil;
+        // if this isn't a dictionary, we don't know what to do so pass it back
+        if ([fieldClause isKindOfClass:[NSDictionary class]] && [fieldClause count] != 0) {
+            fieldName = fieldClause.allKeys[0];
+            predicate = fieldClause[fieldName];
+        } else {
+            [accumulator addObject:fieldClause];
+            continue;
+        }
+        
```

Comment:
The comment above the if statement is confusing me a bit, it implies the first condition will be passing the object back if it is not a dictionary, where as the code shows otherwise, I think it should be more clear what is going on.